src/add_service.rs

```diff
@@ -357,7 +375,130 @@ mod tests {
+
+        let temp_dir = assert_fs::TempDir::new()?;
+        let node_data_dir = temp_dir.child("safenode1");
+        node_data_dir.create_dir_all()?;
+        let node_logs_dir = temp_dir.child("logs");
+        node_logs_dir.create_dir_all()?;
+        let safenode_download_path = temp_dir.child(SAFENODE_FILE_NAME);
+        safenode_download_path.write_binary(b"fake safenode bin")?;
+
+        let custom_port = 12000;
+        let custom_rpc_port = 12001;
+
+        let result = add(
+            AddServiceOptions {
+                genesis: true,
+                count: None,
+                safenode_dir_path: temp_dir.to_path_buf(),
+                service_data_dir_path: node_data_dir.to_path_buf(),
+                service_log_dir_path: node_logs_dir.to_path_buf(),
+                peers: vec![],
+                port: Some(custom_port),
+                rpc_port: Some(custom_rpc_port),
+                url: None,
+                user: get_username(),
+                version: None,
+            },
+            &mut node_registry,
+            &mock_service_control,
+            Box::new(MockSafeReleaseRepository::new()),
+        )
+        .await;
+
+        match result {
+            Ok(_) => panic!("This test should result in an error"),
+            Err(e) => {
+                assert_eq!(format!("A genesis node already exists"), e.to_string())
+            }
+        }
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn add_genesis_node_should_return_an_error_if_count_is_greater_than_1() -> Result<()> {
+        let mock_service_control = MockServiceControl::new();
+
+        let mut node_registry = NodeRegistry {
+            save_path: PathBuf::new(),
+            nodes: vec![],
+            faucet_pid: None,
+        };
+
+        let temp_dir = assert_fs::TempDir::new()?;
+        let node_data_dir = temp_dir.child("safenode1");
+        node_data_dir.create_dir_all()?;
+        let node_logs_dir = temp_dir.child("logs");
+        node_logs_dir.create_dir_all()?;
+        let safenode_download_path = temp_dir.child(SAFENODE_FILE_NAME);
+        safenode_download_path.write_binary(b"fake safenode bin")?;
+
+        let custom_port = 12000;
+        let custom_rpc_port = 12001;
+
+        let result = add(
+            AddServiceOptions {
+                genesis: true,
+                count: Some(3),
+                safenode_dir_path: temp_dir.to_path_buf(),
+                service_data_dir_path: node_data_dir.to_path_buf(),
+                service_log_dir_path: node_logs_dir.to_path_buf(),
+                peers: vec![],
+                port: Some(custom_port),
+                rpc_port: Some(custom_rpc_port),
+                url: None,
+                user: get_username(),
+                version: None,
+            },
+            &mut node_registry,
+            &mock_service_control,
+            Box::new(MockSafeReleaseRepository::new()),
+        )
+        .await;
+
+        match result {
```

Comment:
Not sure if it's more concise, but perhaps you would like this more:

```suggestion
        assert_eq!(Err("A genesis node already exists".to_string()), result.map_err(|e| e.to_string()));
```

`format!(...)` is superfluous here by the way.